File: TestP1/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author zapi24
 */
public class Weapon extends CombatElement {
    
    
    public Weapon(float power, int uses){
        
        super(power,uses);
    }
    
    public float attack(){
        
        return super.produceEffect(); //Utiliza el metodo heredado de CombatElement
    }
    
    @Override
    public String toString(){
        
         return "W" + super.toString();
    }
    
}

File: Irrgarten_ruby/weapon.rb
# frozen_string_literal: true

module Irrgarten

  require_relative 'combat_element'
  class Weapon < CombatElement

    def initialize(power,uses)

      super(power,uses)
    end

    def self.new_weapon(power,uses) #Constructor

      new(power,uses)
    end

    def attack

      produce_effect
    end

    def to_s

      "W #{super.to_s}"
    end
  end

end
